core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala

```diff
@@ -340,6 +340,70 @@ class WorkspaceApiServiceSpec extends ApiServiceSpec {
     runMultipleAndWait(100)(generator)
   }
 
+  it should "return 201 for post to workspaces with onlyAddBillingProjectOwner" in withTestDataApiServices { services =>
+    val newWorkspace = WorkspaceRequest(
+      namespace = testData.wsName.namespace,
+      name = "newWorkspace",
+      Map.empty,
+      onlyAddBillingProjectOwner = Option(true)
+    )
+
+    Post(s"/workspaces", httpJson(newWorkspace)) ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.Created, responseAs[String]) {
+          status
+        }
+        assertResult(newWorkspace) {
+          val ws = runAndWait(workspaceQuery.findByName(newWorkspace.toWorkspaceName)).get
+          WorkspaceRequest(ws.namespace, ws.name, ws.attributes, Option(Set.empty), onlyAddBillingProjectOwner = Option(true))
+        }
+        assertResult(newWorkspace) {
+          val ws = responseAs[WorkspaceDetails]
+          WorkspaceRequest(ws.namespace, ws.name, ws.attributes.getOrElse(Map()), Option(Set.empty), onlyAddBillingProjectOwner = Option(true))
+        }
+        // TODO: does not test that the path we return is correct.  Update this test in the future if we care about that
+        assertResult(Some(Location(Uri("http", Uri.Authority(Uri.Host("example.com")), Uri.Path(newWorkspace.path))))) {
+          header("Location")
+        }
+      }
+  }
+
+  it should "return 403 for post to workspaces with onlyAddBillingProjectOwner without BP owner permissions" in withTestDataApiServicesMockitoSam { services =>
+    val newWorkspace = WorkspaceRequest(
+      namespace = testData.wsName.namespace,
+      name = "newWorkspace",
+      Map.empty,
+      onlyAddBillingProjectOwner = Option(true)
+    )
+
+    // User has BP user role, but not owner role
+    when(services.samDAO.listUserRolesForResource(
+      ArgumentMatchers.eq(SamResourceTypeNames.billingProject),
+      ArgumentMatchers.eq(testData.billingProject.projectName.value),
+      any[UserInfo]
+    )).thenReturn(Future.successful(Set[SamResourceRole](SamProjectRoles.workspaceCreator, SamProjectRoles.batchComputeUser)))
```

Comment:
I added `SamProjectRoles.batchComputeUser` because `addUserToBillingProject` adds the user to `SamBillingProjectPolicyNames.canComputeUser` :

```
      case ProjectRoles.User => Seq(SamBillingProjectPolicyNames.workspaceCreator, SamBillingProjectPolicyNames.canComputeUser)
```